### PR TITLE
Improve performance of `bracket` for `AbstractLieAlgebraElem`

### PR DESCRIPTION
**Pull request by lgoettgens**
by being a bit smarter with handling sparse rows

### CODE REVIEW

File: experimental/LieAlgebras/src/AbstractLieAlgebra.jl
Comment (fingolfin) [inline experimental/LieAlgebras/src/AbstractLieAlgebra.jl:76]: I double checked and it seems this is indeed the correct argument order -- very confusing when one already knows `addmul!` :-(. 

On the up side, perhaps we can just simply define methods
```julia
addmul!(a::SRow{T}, b::SRow{T}, c::T) where T = add_right_scaled_row(b, a, c) 
addmul!(a::SRow{T}, b::T, c::SRow{T}) where T = add_left_scaled_row(c, a, b)
```
and migrate to those over time?


But of course that's not any reason to hold up this PR, just musings :-)

Comment (lgoettgens) [reply experimental/LieAlgebras/src/AbstractLieAlgebra.jl:76]: See https://github.com/thofma/Hecke.jl/pull/1659. I would suggest to merge this PR here right now and then once https://github.com/thofma/Hecke.jl/pull/1659 is available go through Oscar and phase out the `add_scaled_row!` calls. (if you agree please approve and merge this)

**Review by fingolfin — APPROVED** [review]

**Pull request merged**
